runner/src/server/plugins/engine/pluginHandlers/files/prehandlers/handleUpload.ts

```diff
@@ -108,9 +102,13 @@ export async function handleUpload(
         loggerIdentifier,
         `Document upload API responded with an error ${error}`
       );
+      errors.add(error);
```

Comment:
I think this can be omitted, since it's handled in l51-53. I think that should remove the need for the set as well?